Modules/Core/Common/include/itkMacro.h

```diff
@@ -441,62 +441,73 @@ OutputWindowDisplayDebugText(const char *);
 #  define ITK_LOCATION "unknown"
 #endif
 
-/** The exception macro is used to print error information (i.e., usually
- * a condition that results in program failure). Example usage looks like:
- * itkExceptionMacro(<< "this is error info" << this->SomeVariable); */
-#define itkExceptionMacro(x)                                                                                           \
-  {                                                                                                                    \
-    std::ostringstream message;                                                                                        \
-    message << "itk::ERROR: " << this->GetNameOfClass() << "(" << this << "): " x;                                     \
-    ::itk::ExceptionObject e_(__FILE__, __LINE__, message.str().c_str(), ITK_LOCATION);                                \
-    throw e_; /* Explicit naming to work around Intel compiler bug.  */                                                \
-  }
-
-#define itkGenericExceptionMacro(x)                                                                                    \
-  {                                                                                                                    \
-    std::ostringstream message;                                                                                        \
-    message << "itk::ERROR: " x;                                                                                       \
-    ::itk::ExceptionObject e_(__FILE__, __LINE__, message.str().c_str(), ITK_LOCATION);                                \
-    throw e_; /* Explicit naming to work around Intel compiler bug.  */                                                \
-  }
-
 #define itkDeclareExceptionMacro(newexcp, parentexcp, whatmessage)                                                     \
```

Comment:
NOTE:  itkDeclareExceptionMacro is ONLY used in 1 class in ITK.  A search of Slicer, ANTs, BRAINSTools, all remote modules shows that this macro has not been used anywhere else.

Reply:
If it is also not used in ITKExamples and software guide, maybe we should deprecate it now?